services/galley/src/Galley/Intra/BackendNotificationQueue.hs

```diff
@@ -78,6 +80,18 @@ enqueueNotificationsConcurrently m xs f = do
     qualifyAs r
       <$> enqueueSingleNotification (tDomain r) m chanVar (f r)
 
+enqueueNotificationsConcurrentlyBuckets ::
+  (Foldable f) =>
+  Q.DeliveryMode ->
+  f (Remote x) ->
+  (Remote x -> FedQueueClient c a) ->
+  ExceptT FederationError App [Remote a]
+enqueueNotificationsConcurrentlyBuckets m xs f = do
+  chanVar <- getChannel
+  lift $ pooledForConcurrentlyN 8 (toList xs) $ \r ->
+    qualifyAs r
+      <$> enqueueSingleNotification (tDomain r) m chanVar (f r)
+
```

Comment:
Looks wrong, the other one is bucketing, not this one.

Reply:
This is named according to the scheme used by synchronous notifications. This function assumes that the inputs are already bucketed by remote. The other one does the bucketing automatically.